src/main/java/com/petros/bibernate/datasource/BibernateConnection.java

```diff
@@ -6,13 +6,13 @@
 import java.sql.*;
 import java.util.Map;
 import java.util.Properties;
-import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.Queue;
 import java.util.concurrent.Executor;
 
 @RequiredArgsConstructor
 public class BibernateConnection implements Connection {
     private final Connection connection;
-    private final ConcurrentLinkedQueue<BibernateConnection> connectionPool;
+    private final Queue<BibernateConnection> connectionPool;
```

Comment:
Why did you switch to a regular Queue? Our connections will be taken from different threads for different sessions, right?

Reply:
Queue is an interface, implementation - LinkedBlockingQueue